server/src/main/java/io/littlehorse/server/streams/stores/BaseStoreImpl.java

```diff
@@ -14,25 +15,33 @@
 abstract class BaseStoreImpl extends ReadOnlyBaseStoreImpl implements BaseStore {
 
     private final KeyValueStore<String, Bytes> nativeStore;
+    private final MetadataCache metadataCache;
 
-    BaseStoreImpl(KeyValueStore<String, Bytes> nativeStore, TenantIdModel tenantId, ExecutionContext context) {
-        super(nativeStore, tenantId, context);
+    BaseStoreImpl(
+            KeyValueStore<String, Bytes> nativeStore,
+            TenantIdModel tenantId,
+            ExecutionContext context,
+            MetadataCache metadataCache) {
+        super(nativeStore, tenantId, context, metadataCache);
         this.nativeStore = nativeStore;
+        this.metadataCache = metadataCache;
     }
 
-    BaseStoreImpl(KeyValueStore<String, Bytes> nativeStore, ExecutionContext context) {
-        this(nativeStore, null, context);
+    BaseStoreImpl(KeyValueStore<String, Bytes> nativeStore, ExecutionContext context, MetadataCache metadataCache) {
+        this(nativeStore, null, context, metadataCache);
     }
 
     @Override
     public void put(Storeable<?> thing) {
         String key = maybeAddTenantPrefix(thing.getFullStoreKey());
+        metadataCache.removeMissingKey(key);
```

Comment:
Suggestion: instead of removing the missing key, why not overwrite it with the value?